homeassistant/auth/mfa_modules/insecure_example.py

```diff
@@ -0,0 +1,84 @@
+"""Example auth module."""
+import logging
+from collections import OrderedDict
+
+import voluptuous as vol
+
+from . import MultiFactorAuthModule, MULTI_FACTOR_AUTH_MODULES, \
+    MULTI_FACTOR_AUTH_MODULE_SCHEMA
+
+CONFIG_SCHEMA = MULTI_FACTOR_AUTH_MODULE_SCHEMA.extend({
+    vol.Required('users'): [vol.Schema({
+        vol.Required('user_id'): str,
+        vol.Required('pin'): str,
+    })]
+}, extra=vol.PREVENT_EXTRA)
+
+STORAGE_VERSION = 1
+STORAGE_KEY = 'mfa_modules.insecure_example'
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@MULTI_FACTOR_AUTH_MODULES.register('insecure_example')
+class InsecureExampleModule(MultiFactorAuthModule):
+    """Example auth module validate pin."""
+
+    DEFAULT_TITLE = 'Insecure Personal Identify Number'
+
+    def __init__(self, hass, config):
+        """Initialize the user data store."""
+        super().__init__(hass, config)
+        self._data = None
+        self._users = config['users']
+
+    @property
+    def input_schema(self):
+        """Validate login flow input data."""
+        schema = OrderedDict()
+        schema['pin'] = str
+        return vol.Schema(schema)
+
+    @property
+    def setup_schema(self):
+        """Validate async_setup_user input data."""
+        schema = OrderedDict()
+        schema['pin'] = str
+        return vol.Schema(schema)
+
+    async def async_setup_user(self, user_id, data=None):
```

Comment:
I don't see how this can be `None`, at least it's a dict if no fields specified in setup schema